components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -115,4 +114,8 @@ android.applicationVariants.all { variant ->
     // The added directory doesn't appear in the paths listed by the
     // `sourceSets` task, for reasons unknown.
     variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+}}
+
+if (android.hasProperty('applicationVariants')) {
+    ext.gleanGenerateMetricsAPI(android.applicationVariants.all)
```

Comment:
This is where the library-local metrics.yaml for the baseline telemetry will go (in a future PR).

Reply:
Sorry for being pedantic about this 🤦‍♂️ But why is this change needed? (along with all the changes in this file)

For installing the environment, we just need to include the `plugins` and `sdk_generator.gradle` in our gradle file, right? If these changes are not strictly required for this PR, I'd be for dropping them: it would make it easier to reason about them in the follow-up PR related to the library `metrics.yaml`.

Reply:
When using this plugin in the context of the library, there is to `android.applicationVariants` (makes sense, because we aren't building an application in that context), so line 77 of the original gives an "undefined variable" exception.  This turns the chunk that runs the generator into a function so it can be run only when building the application, and in a future PR, it will be called differently to handle the needs of the library.  And it allows the rest of the file (the setting up of the Python environment) to be shared in both the library and application contexts.

Reply:
Ah, gotcha! Thanks for the clarification, it makes sense!